### ISSUE TITLE
No longer available with Python 3, use mutagen.mp4

### ISSUE THREAD

**Issue by marianolatorre** [jrnewell/spotify-ripper#53]
Hi all! I followed all the instruccions to install using brew on mac. Everything seemed ok during installation but the first time I run the script I get this output error:

```
$ spotify-ripper -u myuser -p pass spotify:user:user:playlist:6J6cDk4UVEPMR1rwCW06WS
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.4/bin/spotify-ripper", line 9, in <module>
    load_entry_point('spotify-ripper==2.0.3', 'console_scripts', 'spotify-ripper')()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pkg_resources/__init__.py", line 552, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2672, in load_entry_point
    return ep.load()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2345, in load
    return self.resolve()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pkg_resources/__init__.py", line 2351, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Library/Frameworks/Python.framework/Versions/3.4/bin/main.py", line 7, in <module>
    from spotify_ripper.ripper import Ripper
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify_ripper/ripper.py", line 8, in <module>
    from spotify_ripper.tags import set_metadata_tags
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify_ripper/tags.py", line 6, in <module>
    from mutagen import mp3, id3, flac, oggvorbis, oggopus, m4a, mp4, aac
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/mutagen/m4a.py", line 11, in <module>
    raise ImportError("No longer available with Python 3, use mutagen.mp4")
ImportError: No longer available with Python 3, use mutagen.mp4

$ python --version
Python 2.7.5
```

Error from mutagen indicates something about python 3, but as you can see I'm still using 2.7.5...but also there's all those paths "/Library/Frameworks/Python.framework/Versions/3.4" so I'm a little confussed...if someone could hint me pls! 

thanks

**Comment by jrnewell** [comment]
I am not sure why it says you are using Python 3, but I am will try to improve Python 3 compatibility in the next days.  I was neglecting it lately.

**Comment by marianolatorre** [comment]
thanks

**Comment by jrnewell** [comment]
I tried updating Python 3 compatibility with version ``2.0.4``.  Can you upgrade and see if it works for you?

**Comment by marianolatorre** [comment]
ok, I tried the following but I'm still having some issues:

```
$  pip install --upgrade spotify-ripper
Collecting spotify-ripper
  Downloading spotify-ripper-2.1.3.tar.gz
Requirement already up-to-date: pyspotify==2.0.0b4 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from spotify-ripper)
Requirement already up-to-date: colorama>=0.3.3 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from spotify-ripper)
Requirement already up-to-date: mutagen==1.29 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from spotify-ripper)
Requirement already up-to-date: requests>=2.3.0 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from spotify-ripper)
Requirement already up-to-date: schedule>=0.3.1 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from spotify-ripper)
Requirement already up-to-date: cffi>=0.7 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from pyspotify==2.0.0b4->spotify-ripper)
Requirement already up-to-date: pycparser in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages (from cffi>=0.7->pyspotify==2.0.0b4->spotify-ripper)
Installing collected packages: spotify-ripper
  Found existing installation: spotify-ripper 2.0.3
    Uninstalling spotify-ripper-2.0.3:
      Successfully uninstalled spotify-ripper-2.0.3
  Running setup.py install for spotify-ripper
Successfully installed spotify-ripper-2.1.3
```

and then when running it:
```
Spotify Ripper - v2.1.3
  Encoding output:	MP3, VBR 0
  Spotify bitrate:	320 kbps
  Unicode support:	Yes
  Output directory:	/Users/user/Documents/playlist
  Settings directory:	/Users/user/.spotify-ripper
  Format String:	{album_artist}/{album}/{artist} - {track_name}.{ext}
  Overwrite files:	No
Logging in...
Logged in as user
Loading playlist...
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify_ripper/progress.py", line 60, in calc_total
    track.load(timeout=2)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify/track.py", line 80, in load
    return utils.load(self._session, self, timeout=timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify/utils.py", line 240, in load
    raise spotify.Timeout(timeout)
spotify.error.Timeout: Operation did not complete in 2.000s

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify_ripper/ripper.py", line 139, in run
    self.progress.calc_total(tracks)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/spotify_ripper/progress.py", line 67, in calc_total
    except spotify.Error as e:
NameError: name 'spotify' is not defined
```

**Comment by marianolatorre** [comment]
ehm...and now i's working :) (I didn't change anything)

**Comment by jrnewell** [comment]
It looks like it was handling that time-out exception gracefully.  This should be fixed in ``2.2.0``.
